tests/fixtures/saas_fixtures.py

```diff
@@ -109,3 +227,29 @@ def mailchimp_account_email():
     return pydash.get(saas_config, "mailchimp.account_email") or os.environ.get(
         "MAILCHIMP_ACCOUNT_EMAIL"
     )
+
+
+@pytest.fixture(scope="function")
+def reset_saas_data(connection_config_saas, mailchimp_account_email) -> Generator:
+    """
+    Gets the current value of the resource and restores it after the test is complete.
+    Used for erasure tests.
+    """
+    connector = SaaSConnector(connection_config_saas)
+    request = (
+        "GET",
+        "/3.0/search-members",
+        {"query": mailchimp_account_email},
+        {},
+    )
+    response = connector.create_client().send(request)
+    body = response.json()
+    member = body["exact_matches"]["members"][0]
+    yield member
+    request = (
+        "PUT",
+        f'/3.0/lists/{member["list_id"]}/members/{member["id"]}',
```

Comment:
I might well be missing something here — why is it useful to add the member to a list in the teardown on this fixture?

In order to keep this as idempotent as we can, would it be better to add the data we need to Mailchimp in the setup, then remove it again in the teardown? We might also want to use `session` scoped fixtures for this instead of `function` to avoid making these API calls at the start and end of every test.

Reply:
At a high level, we're reading the current `member` value from Mailchimp and restoring it in the teardown. I'm using this as part of every Mailchimp erasure to restore the original value after asserting the erasure works. I also did this to avoid having to specify `list_id` and `id` in the test, using this approach we can just read it from the `GET` request and use those values to put the member back in the specific location in the teardown.

Reply:
I can see why this approach would be appealing, because it's the lightest touch on the external API possible. I'd worry that it relying on data being present on Mailchimp leaves it susceptible to interference — any change in the data on Mailchimp will affect whether any test using this fixture passes or fails.